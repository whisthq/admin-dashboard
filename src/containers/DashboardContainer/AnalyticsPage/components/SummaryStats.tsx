import React from 'react'
import { connect } from 'react-redux'
import { Card, Row, Col } from 'antd'

import 'antd/dist/antd.css'

import { fetchTotalSignups, fetchTotalMinutes } from '../../../../actions/index'

class SummaryStats extends React.Component<any, any> {
    componentDidMount() {
        this.props.dispatch(fetchTotalSignups())
        this.props.dispatch(fetchTotalMinutes())
    }

    render() {
        return (
            <div>
                <p
                    style={{
                        fontWeight: 'bold',
                        fontSize: 24,
                    }}
                >
                    Summary
                </p>
                <Row gutter={16}>
                    <Col span={8}>
                        <Card
                            title="Total Session length"
                            size="small"
                            loading={this.props.totalMinutes == null}
                        >
                            {this.props.totalMinutes !== null && (
                                <div>
                                    <p>
                                        Past day:{' '}
                                        <strong>
                                            {Math.round(
                                                this.props.totalMinutes['day'] /
                                                    60
                                            ).toString() +
                                                'h ' +
                                                (
                                                    this.props.totalMinutes[
                                                        'day'
                                                    ] % 60
                                                ).toString() +
                                                'm'}
                                        </strong>
                                    </p>
                                    <p>
                                        Past week:{' '}
                                        <strong>
                                            {Math.round(
                                                this.props.totalMinutes[
                                                    'week'
                                                ] / 60
                                            ).toString() +
                                                'h ' +
                                                (
                                                    this.props.totalMinutes[
                                                        'week'
                                                    ] % 60
                                                ).toString() +
                                                'm'}
                                        </strong>
                                    </p>
                                    <p>
                                        Past month:{' '}
                                        <strong>
                                            {Math.round(
                                                this.props.totalMinutes[
                                                    'month'
                                                ] / 60
                                            ).toString() +
                                                'h ' +
                                                (
                                                    this.props.totalMinutes[
                                                        'month'
                                                    ] % 60
                                                ).toString() +
                                                'm'}
                                        </strong>
                                    </p>
                                </div>
                            )}
                        </Card>
                    </Col>
                    <Col span={8}>
                        <Card
                            title="Number of signups"
                            size="small"
                            loading={this.props.totalSignups == null}
                        >
                            {this.props.totalSignups !== null && (
                                <div>
                                    <p>
                                        Past day:{' '}
                                        <strong>
                                            {this.props.totalSignups['day']}
                                        </strong>
                                    </p>
                                    <p>
                                        Past week:{' '}
                                        <strong>
                                            {this.props.totalSignups['week']}
                                        </strong>
                                    </p>
                                    <p>
                                        Past month:{' '}
                                        <strong>
                                            {this.props.totalSignups['month']}
                                        </strong>
                                    </p>
                                </div>
                            )}
                        </Card>
                    </Col>
                </Row>
            </div>
        )
    }
}

function mapStateToProps(state: any) {
    return {
        totalSignups: state.AccountReducer.totalSignups,
        totalMinutes: state.AccountReducer.totalMinutes,
    }
}

export default connect(mapStateToProps)(SummaryStats)
