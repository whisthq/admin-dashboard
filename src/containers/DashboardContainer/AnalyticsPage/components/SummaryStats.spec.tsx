import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Card, Row, Col } from 'antd'
import toJson from 'enzyme-to-json'

import { SummaryStats } from './SummaryStats'

// necessary for enzyme to work
Enzyme.configure({ adapter: new Adapter() })

/*
SummaryStats should tell us about some summary stats (like total number of minutes, of signups, etcetera)
about customers and so forth. It will have one row with two columns. Each column has a card with a chunk of the
summary stats (one is minutes, one is signups).
*/
describe('<SummaryStats />', () => {
    // snapshot test
    it('renders correctly', () => {
        const wrapper = shallow(
            <SummaryStats
                totalMinutes={{
                    day: 0,
                }}
                totalSignups={{
                    day: 0,
                }}
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )

        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('componentDidMount fetches both minutes and signups', () => {
        // TODO
    })

    it('has one row and two columns and two cards', () => {
        const wrapper = shallow(
            <SummaryStats
                totalMinutes={{
                    day: 0,
                }}
                totalSignups={{
                    day: 0,
                }}
                dispatch={(items: any) => {
                    // mock
                }}
            />
        )
        expect(wrapper.find(Row)).toHaveLength(1)
        expect(wrapper.find(Col)).toHaveLength(2)
        expect(wrapper.find(Card)).toHaveLength(2)
    })
})
