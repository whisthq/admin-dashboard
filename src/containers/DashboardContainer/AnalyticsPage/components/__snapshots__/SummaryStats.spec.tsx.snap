// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryStats /> renders correctly 1`] = `
<div>
  <p
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Summary
  </p>
  <Row
    gutter={16}
  >
    <Col
      span={8}
    >
      <Card
        loading={false}
        size="small"
        title="Total Session length"
      >
        <div>
          <p>
            Past day:
             
            <strong>
              0h 0m
            </strong>
          </p>
          <p>
            Past week:
             
            <strong>
              NaNh NaNm
            </strong>
          </p>
          <p>
            Past month:
             
            <strong>
              NaNh NaNm
            </strong>
          </p>
        </div>
      </Card>
    </Col>
    <Col
      span={8}
    >
      <Card
        loading={false}
        size="small"
        title="Number of signups"
      >
        <div>
          <p>
            Past day:
             
            <strong>
              0
            </strong>
          </p>
          <p>
            Past week:
             
            <strong />
          </p>
          <p>
            Past month:
             
            <strong />
          </p>
        </div>
      </Card>
    </Col>
  </Row>
</div>
`;
